linode/objbucket/resource.go

```diff
@@ -157,19 +165,28 @@ func updateResource(
 	lifecycleChanged := d.HasChange("lifecycle_rule")
 
 	if versioningChanged || lifecycleChanged {
+		endpoint := d.Get("endpoint").(string)
```

Comment:
`endpoint` might not be populated on resource creation because `updateResource` is called directly from `createResource` before `readResource` is called. To work around this, you could set `endpoint` in the state [here](https://github.com/linode/terraform-provider-linode/blob/ad33b5b1ea82cdbc23506ee6aa99406ba55d2e52/linode/objbucket/resource.go#L141) to ensure it's populated at creation.

FWIW this resource's implementation is super hacky as a whole, so it'd probably be worth refactoring in the future :slightly_smiling_face: